File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/domain/entity/GenreEntity.java
package ru.spring5.elena.librarymanager.domain.entity;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import static javax.persistence.GenerationType.IDENTITY;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "GENERES")
@Getter
@NoArgsConstructor
//@RequiredArgsConstructor
@Setter
public class GenreEntity {
    @Id
    @GeneratedValue(strategy = IDENTITY) // уникальный ID, генерируется сам в БД
    @Column(name = "GENERE_ID")
    @NotNull
    @Setter(AccessLevel.NONE)
    private Long genreId;

    @Version
    @Column(name = "VERSION")
    private int version; //версия, инкрементится при редактировании записи

    @Column(name = "TITLE")
    private String title; //
    
    @OneToMany(mappedBy = "genre", cascade = CascadeType.ALL, orphanRemoval = true)
    @Setter(AccessLevel.NONE)
    private Set<EditionEntity> editions = new HashSet<EditionEntity>();
}

File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/domain/entity/GenreEntity.kt
package ru.elenacherev.librarymanager.domain.entity

import org.hibernate.annotations.ColumnDefault
import java.util.*
import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table
import javax.persistence.Version
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@Table(name = "GENRES")
data class GenreEntity(

    @Column(name = "GENRE_ID", nullable = false, updatable = false)
    @GeneratedValue
    @Id
    @get:NotNull
    var genreId: UUID? = null,

    //версия, инкрементится при редактировании записи
    @Version
    @Column(name = "VERSION", nullable = false)
    var version: Int = 0,

    @Column(name = "TITLE")
    @Size(min = 1, max = 255)
    var title: String,

    @OneToMany(mappedBy = "genre", cascade = [CascadeType.ALL], orphanRemoval = true)
    val editions: MutableSet<EditionEntity> = mutableSetOf()
)